vcr/pe/presentation_definition.go

```diff
@@ -473,70 +473,81 @@ func getValueAtPath(path string, vcAsInterface interface{}) (interface{}, error)
 // Supported go value types: string, float64, int, bool and array.
 // 'null' values are not supported.
 // It returns an error on unsupported features or when the regex pattern fails.
-func matchFilter(filter Filter, value interface{}) (bool, error) {
+func matchFilter(filter Filter, value interface{}) (bool, interface{}, error) {
 	// first we check if it's an enum, so we can recursively call matchFilter for each value
 	if filter.Enum != nil {
 		for _, enum := range filter.Enum {
 			f := Filter{
 				Type:  "string",
 				Const: &enum,
 			}
-			match, _ := matchFilter(f, value)
+			match, result, _ := matchFilter(f, value)
 			if match {
-				return true, nil
+				return true, result, nil
 			}
 		}
-		return false, nil
+		return false, nil, nil
 	}
 
 	switch typedValue := value.(type) {
 	case string:
 		if filter.Type != "string" {
-			return false, nil
+			return false, nil, nil
 		}
 	case float64:
 		if filter.Type != "number" {
-			return false, nil
+			return false, nil, nil
 		}
 	case int:
 		if filter.Type != "number" {
-			return false, nil
+			return false, nil, nil
 		}
 	case bool:
 		if filter.Type != "boolean" {
-			return false, nil
+			return false, nil, nil
 		}
 	case []interface{}:
 		for _, v := range typedValue {
-			match, err := matchFilter(filter, v)
+			match, _, err := matchFilter(filter, v)
 			if err != nil {
-				return false, err
+				return false, nil, err
 			}
 			if match {
-				return true, nil
+				return true, value, nil
```

Comment:
should this return `v`?

Reply:
Before, it returned the whole slice if a slice entry matches. I wanted to retain that behavior (therefore, `value`, which is the whole slice).